azure-pipelines.yml

```
@@ -130,22 +134,218 @@ jobs:
+      Contents: |
+          Microsoft.Management.Configuration\Microsoft.Management.Configuration.dll
+      TargetFolder: $(buildOutDirAnyCpu)\PowerShell\Microsoft.WinGet.Configuration\SharedDependencies\$(BuildPlatform)
+      flattenFolders: true
+
+  - task: CopyFiles@2
+    displayName: 'Copy managed binaries for Microsoft.WinGet.Configuration in arch specific'
+    inputs:
+      SourceFolder: $(buildOutDirAnyCpu)
+      Contents: |
+          Microsoft.Management.Configuration.Projection\net6.0-windows10.0.22000.0\Microsoft.Management.Configuration.Projection.dll
+      TargetFolder: $(buildOutDirAnyCpu)\PowerShell\Microsoft.WinGet.Configuration\SharedDependencies\$(BuildPlatform)
+      flattenFolders: true
+
+  - task: CopyFiles@2
+    displayName: 'Copy PowerShell AnyCPU Module Files'
+    inputs:
+      SourceFolder: '$(buildOutDirAnyCpu)\PowerShell'
+      TargetFolder: '$(artifactsDir)\PowerShell'
+
+  - task: CopyFiles@2
+    displayName: 'Copy E2E test binaries'
+    inputs:
+      SourceFolder: '$(buildOutDir)\AppInstallerCLIE2ETests'
+      TargetFolder: '$(artifactsDir)\AppInstallerCLIE2ETests'
+
+  - task: CopyFiles@2
+    displayName: 'Copy Unit test binaries'
+    inputs:
+      SourceFolder: '$(buildOutDir)\AppInstallerCLITests'
+      TargetFolder: '$(artifactsDir)\AppInstallerCLITests'
+
+  - task: CopyFiles@2
+    displayName: 'Copy LocalhostWebServer binaries'
+    inputs:
+      SourceFolder: '$(buildOutDir)\LocalhostWebServer'
+      TargetFolder: '$(artifactsDir)\LocalhostWebServer'
+
+  - task: CopyFiles@2
+    displayName: 'Copy Files: WinGetUtilInterop.UnitTests'
+    inputs:
+      SourceFolder: '$(Build.SourcesDirectory)\src\WinGetUtilInterop.UnitTests\bin\$(buildPlatform)\$(BuildConfiguration)\net6.0'
+      TargetFolder:  '$(artifactsDir)\WinGetUtilInterop.UnitTests\'
+      CleanTargetFolder: true
+      OverWrite: true
+
+  - task: CopyFiles@2
+    displayName: 'Copy WinGetUtil to WinGetUtilInterop.UnitTests folder'
+    inputs:
+      Contents: |
+          $(buildOutDir)\WinGetUtil\WinGetUtil.dll
+      TargetFolder: '$(artifactsDir)\WinGetUtilInterop.UnitTests\'
+      flattenFolders: true
+
+  - task: CopyFiles@2
+    displayName: 'Copy Microsoft.Management.Configuration binaries'
+    inputs:
+      SourceFolder: '$(buildOutDir)\Microsoft.Management.Configuration'
+      TargetFolder: '$(artifactsDir)\Microsoft.Management.Configuration'
+
+  - task: CopyFiles@2
+    displayName: 'Copy Microsoft.Management.Configuration.UnitTests binaries'
+    inputs:
+      SourceFolder: '$(buildOutDir)\Microsoft.Management.Configuration.UnitTests'
+      TargetFolder: '$(artifactsDir)\Microsoft.Management.Configuration.UnitTests'
+
+  - task: CopyFiles@2
+    displayName: 'Copy Microsoft.Management.Configuration.OutOfProc binaries'
+    inputs:
+      SourceFolder: '$(buildOutDir)\Microsoft.Management.Configuration.OutOfProc'
+      TargetFolder: '$(artifactsDir)\Microsoft.Management.Configuration.OutOfProc'

```

I think many of these could be done in a single step

```suggestion
  - task: CopyFiles@2
    displayName: 'Copy binaries'
    inputs:
      SourceFolder: '$(buildOutDirAnyCpu)'
      TargetFolder: '$(artifactsDir)'
      Contents: |
        PowerShell\**
        AppInstallerCLIE2ETests\**
        AppInstallerCLITests\**
        LocalhostWebServer\**
        Microsoft.Management.Configuration\**
        Microsoft.Management.Configuration.UnitTests\**
        Microsoft.Management.Configuration.OutOfProc\**

  - task: CopyFiles@2
    displayName: 'Copy Files: WinGetUtilInterop.UnitTests'
    inputs:
      SourceFolder: '$(Build.SourcesDirectory)\src\WinGetUtilInterop.UnitTests\bin\$(buildPlatform)\$(BuildConfiguration)\net6.0'
      TargetFolder:  '$(artifactsDir)\WinGetUtilInterop.UnitTests\'
      CleanTargetFolder: true
      OverWrite: true

  - task: CopyFiles@2
    displayName: 'Copy WinGetUtil to WinGetUtilInterop.UnitTests folder'
    inputs:
      Contents: |
          $(buildOutDir)\WinGetUtil\WinGetUtil.dll
      TargetFolder: '$(artifactsDir)\WinGetUtilInterop.UnitTests\'
      flattenFolders: true
```

---

Will apply this after my current attempt is done.